docs/generate.py

```diff
@@ -0,0 +1,206 @@
+import ast
+import importlib
+import inspect
+import pkgutil
+from importlib import import_module
+from pathlib import Path
+from types import ModuleType
+from typing import List
+
+import mkdocs_gen_files
+import streamlit_extras
+
+# TODO: Check why query_string fails
+extra_modules_names = [
+    extra.name
+    for extra in pkgutil.iter_modules(streamlit_extras.__path__)
+    if extra.name != "query_string"
+]
+
+EXTRA_EXAMPLE_MD_TEMPLATE = """
+### `{func_name}`
+``` py
+{func_source}
+```
+"""
+EXTRA_FUNCTIONS_MD_TEMPLATE = """
+### `{func_name}`
+
+#### ::: src.streamlit_extras.{module_name}.{func_name}
+
+**Import:**
+
+``` py
+from streamlit_extras.{module_name} import {func_name} # (1)!
+```
+
+1. You should add this to the top of your .py file :hammer_and_wrench:
+
+"""
+
+EXTRA_MD_TEMPLATE = """
+
+# {pretty_title}
+
+Submitted by **{author}**
+
+## Summary
+{description}
+
+## Functions
+{functions_docstrings}
+
+
+"""
+
+EXTRA_MD_TEMPLATE_IF_PYPI = """
+
+# {pretty_title}
+
+Submitted by **{author}**
+
+## Summary
+{description}
+
+## Docstring
+Visit the [PyPI page](https://pypi.org/project/{pypi_name}/) for more information.
+"""
+
+
+def find_decorated_functions(
+    module_name: str, decorator_name: str = "extra"
+) -> List[str]:
+    """Parses a module to find out which functions are decorated with a given decorator
+    We use this to identify the functions we want to show in the docs, which are
+    decorated with `@extra`.
+
+    Args:
+        module_name (str): Name of the module
+        decorator_name (str, optional): Name of the decorator. Defaults to "extra".
+
+    Returns:
+        List[str]: Decorated functions with @decorator_name
+    """
+    module = importlib.import_module(module_name)
+    decorated_functions = []
+
+    assert module.__file__, f"Module {module_name} has no __file__ attribute"
+
+    with open(module.__file__, "r") as module_file:
+        module_source = module_file.read()
+
+    parsed_module = ast.parse(module_source)
+
+    for node in ast.walk(parsed_module):
+        if isinstance(node, ast.FunctionDef):
+            for decorator in node.decorator_list:
+                if isinstance(decorator, ast.Name) and decorator.id == decorator_name:
+                    decorated_functions.append(node.name)
+
+    return decorated_functions
+
+
+def find_example_functions(module_name: str) -> List[str]:
```

Comment:
Why not just use `__examples__`?

Reply:
100%. Fixed in [f3be42f](https://github.com/arnaudmiribel/streamlit-extras/pull/182/commits/f3be42fd079714afe83e12e1ae058b4a02e0156d) - thanks!